hail/src/main/scala/is/hail/expr/ir/LowerMatrixIR.scala

```diff
@@ -6,49 +6,41 @@ import is.hail.expr.types.virtual.{TArray, TInt32, TInterval}
 import is.hail.utils._
 
 object LowerMatrixIR {
-  val entriesFieldName = MatrixType.entriesIdentifier
```

Comment:
I'm not sure why the diff is so terribly horrible for this file. My changes involved:

 * switching from using partial functions to match statements with catch-all base cases. There is no base case for MatrixIR 🎉 
 * doing the typecheck after these match statements
 * Adding the lowering rule for MatrixMapCols

That's it.